FileSystem/DirectoryManager.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Directory Manager Library
+ */
+var DirectoryManager = {
+  currentDirectory: '',
+  getCurrentDirectory: getCurrentDirectory,
+  setCurrentDirectory: setCurrentDirectory,
+  getAbsolutePath: getAbsolutePath,
+  curDir: curDir,
+  fileSystemType: '',
+  getFileSystemType: getFileSystemType_,
+};
+
+/**
+ * Internal function to get the current directory.
+ * If file directory is not initialized, it is
+ * initialized to current active workbook path.
+ * @return {string} current directory
+ */
+function getCurrentDirectory() {
+  if (this.currentDirectory == '') {
+    // Initialize current directory to active workbook path
+    var path = Workbook.getActiveWorkbookPath();
+    this.setCurrentDirectory(path);
+  }
+  return this.currentDirectory;
+}
+
+/**
+ * Internal function to set the current directory.
+ * @param {string} path Current directory path
+ */
+function setCurrentDirectory(path) {
+  if (!isValidAbsolutePath(path)) {
+    throw new Error(path + ' is an invalid path');
+  }
+  this.currentDirectory = sanitizePath(path);
+}
+
+/**
+ * Function to get absolute path given relative path.
+ * If the relative path is absolute, then the relative path is returned
+ * @param {string} relativePath Relative or Absolute path
```

Comment:
This is confusing, just say relative path. It's clear from the description above that a path can be relative/absolute

Reply:
Fixed